stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -431,7 +432,12 @@ public void onMessage(RespT message) {
             .asRuntimeException();
       }
       firstResponseReceived = true;
-      observer.onNext(message);
+
+      if (streamingResponse) {
+        observer.onNext(message);
+      } else {
+        unaryMessage = message;
+      }
 
       if (streamingResponse && adapter.autoFlowControlEnabled) {
```

Comment:
Nit: could move this into the new if block

Reply:
I added a comment here, I think it is more helpful than move it into a new if block.